icu4c/source/i18n/calendar.cpp

```diff
@@ -4251,17 +4251,20 @@ int32_t Calendar::internalGetMonth(UErrorCode& status) const {
     if (U_FAILURE(status)) {
         return 0;
     }
-    if (resolveFields(kMonthPrecedence) == UCAL_MONTH) {
-        return internalGet(UCAL_MONTH, status);
+    if (resolveFields(kMonthPrecedence) == UCAL_ORDINAL_MONTH) {
+        return internalGet(UCAL_ORDINAL_MONTH);
     }
-    return internalGet(UCAL_ORDINAL_MONTH, status);
+    return internalGet(UCAL_MONTH);
 }
 
-int32_t Calendar::internalGetMonth(int32_t defaultValue, UErrorCode& /* status */) const {
-    if (resolveFields(kMonthPrecedence) == UCAL_MONTH) {
-        return internalGet(UCAL_MONTH, defaultValue);
+int32_t Calendar::internalGetMonth(int32_t defaultValue, UErrorCode& status) const {
+    if (U_FAILURE(status)) {
+        return 0;
+    }
+    if (resolveFields(kMonthPrecedence) == UCAL_ORDINAL_MONTH) {
+        return internalGet(UCAL_ORDINAL_MONTH);
```

Comment:
why not
```suggestion
        return internalGet(UCAL_ORDINAL_MONTH, defaultValue);
```
?

Reply:
```
    inline int32_t internalGet(UCalendarDateFields field, int32_t defaultValue) const {return fStamp[field]>kUnset ? fFields[field] : defaultValue;}
```
The default value is used when the value of that field is unset  the field is UCAL_ORDINAL_MONTH is set here, so there are no point to pass with it.